CMake/RealmCore.cmake

```diff
@@ -215,16 +215,28 @@ function(download_realm_core core_version)
     set_property(TARGET realm-parser PROPERTY IMPORTED_LOCATION ${core_parser_library_release})
 endfunction()
 
+macro(propagate_flag out_var flag)
+    if(NOT ${flag} STREQUAL "")
+        list(APPEND ${out_var} -D${flag}=${${flag}})
+    endif()
+endmacro()
+
 macro(build_realm_core)
     set(core_prefix_directory "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/realm-core")
 
+    set(extra_flags "")
+    propagate_flag(extra_flags CMAKE_CXX_COMPILER)
```

Comment:
I needed to do this in order to be able to test with libc++ on linux. If you'd prefer, I can pull this commit out of this PR and do it separately.

Reply:
I'm not the best person to ask for this.

Reply:
It doesn't matter much since this will be going away with the move to monorepo anyway.

Reply:
👍